src/jit/assertionprop.cpp

```diff
@@ -2995,6 +2995,28 @@ AssertionIndex Compiler::optGlobalAssertionIsEqualOrNotEqual(ASSERT_VALARG_TP as
         {
             return assertionIndex;
         }
+
+        // Look for matching exact type assertions based on vtable accesses
+        if ((curAssertion->assertionKind == OAK_EQUAL) && (curAssertion->op1.kind == O1K_EXACT_TYPE) &&
+            op1->OperIs(GT_IND))
+        {
+            GenTreeIndir* indir = op1->AsIndir();
+
+            if (!indir->HasIndex() && (indir->Offset() == 0))
```

Comment:
I don't think you should use `GenTreeIndir`'s `HasIndex`, `Base` and `Offset`. These are practically meaningless before lowering because they depend on containment and address modes. Before lowering:
* `HasIndex()` is always false
* `Base()` is just `Addr()`
* `Offset()` is always 0
Basically these shouldn't exist on `GenTreeIndir`, they're misleading.

Reply:
Good point, I can remove these.